### ISSUE TITLE
Uncaught (in promise) Error: Loading chunk # failed.

### ISSUE THREAD

**Issue by blasterbug** [webpack/webpack#7502]
# Bug report
`Uncaught (in promise) Error: Loading chunk # failed.` after rolling update.

**What is the current behavior?**
I have a kubernetes cluster with a couple of pods running an image of a web app build with webpack. When I build a new image and deploy it to the pods (rolling update), the web app can't be loaded and I get `error: Uncaught (in promise) Error: Loading chunk <whatever chunk number> failed.` in web browser console after reloading the web app. 

**If the current behavior is a bug, please provide the steps to reproduce.**
Basically, webpack runs on Google container builder. The web app is built with docker. 
Once the docker build is done, the image with the new build of the web app is deployed to kubernetes (rolling update). While new pods are created and running with the new version of the web app can't be loaded.

**What is the expected behavior?**
The web app should load smoothly update after update.

**Other relevant information:**
webpack version:  4.11.1
Node.js version:  8.11
Operating System:  CoreOS
Additional tools: Docker, Kubernetes (on Google Cloud)

**PS**: Actually I am not 100% sure if this is a problem directly link to webpack but a combinaison with kubernetes. As I am writing this bug report, I sort figured out the rolling update might the problem as two  different versions of the web app is living as long all the pods are not using the last image.

**Comment by evilebottnawi** [comment]
@blasterbug thanks for issue, it is hard to debug, can you provide more information? It is works in browsers? Can your provide webpack config file?

**Comment by blasterbug** [comment]
No problem.
Any browser (tested with Chrome, Safary and Firefox on MacOS). 
Webpack config is quite simple: 

**webpack.config.js**
```javascript
'use strict';

const path = require('path');
const webpack = require('webpack');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

// http://webpack.github.io/docs/configuration.html
// http://webpack.github.io/docs/webpack-dev-server.html
module.exports = {
  output: {
    chunkFilename: '[id].[hash].js',
    // https://reactjs.org/docs/cross-origin-errors.html
    crossOriginLoading: "anonymous",
    filename: '[name].[hash].js',
    path: path.resolve(__dirname, 'dist'),
  },
  module: {
    rules: [
      {
        // https://github.com/jtangelder/sass-loader
        test: /\.scss$/,
        loader: [
          'style-loader',
          'css-loader',
          'sass-loader',
        ],
      },
      {
        test: /\.css$/,
        use: [
          MiniCssExtractPlugin.loader,
          'css-loader',
        ],
      },
    ],
  },
  plugins: [
    new MiniCssExtractPlugin({
      filename: "[name].[hash].css",
      chunkFilename: "[id].[hash].css",
    }),
    new HtmlWebpackPlugin({
      cache: true,
      title: 'Loogup',
      template: path.join(__dirname, 'public', 'index.html'),
      favicon: path.join(__dirname, 'public', 'favicon.jpg'),
      description: 'Just copy/paste my actuall stuff here',
      // url: 'whatever',
      output:path.resolve(__dirname, 'dist'),
      xhtml: true,
    }),
  ]
};
```   

**webpack.config.prod.js**
```javascript
'use strict';

const path = require('path');
const webpack = require('webpack');
const merge = require('webpack-merge');
const config = require('./webpack.config.js');
const CleanWebpackPlugin = require('clean-webpack-plugin');
const CompressionPlugin = require('compression-webpack-plugin');
const CopyWebpackPlugin = require('copy-webpack-plugin');
const FaviconsWebpackPlugin = require('favicons-webpack-plugin');
const UglifyJsPlugin = require('uglifyjs-webpack-plugin');
const googleMapsApiKey = require('./secrets/googleMapsApiKey.js');

module.exports = merge(
  config,
  {
    mode: 'production',
    entry: {
      app: path.resolve(__dirname, 'src/index.js'),
      admin: path.resolve(__dirname, 'src/components/AdminDashboard'),
      libs: [
        path.resolve(__dirname, 'src/libs'),
        path.resolve(__dirname, 'src/components/LuIcons'),
      ],
      vendor: [
        'prop-types',
        'react',
        'react-dom',
        'react-intl',
        'react-router',
        'redux',
      ],
    },
    output: {
      publicPath: '/', // FIX: Uncaught (in promise) Error: Loading chunk # failed. async
    },
    module: {
      rules: [
        {
          test: /\.js$/,
          exclude: [
            /node_modules/,
            /secrets/ ,
          ],
          use: [
            'babel-loader'
          ],
        },
      ],
    },
    plugins: [
      new CleanWebpackPlugin(
        'dist',
        {
          root: path.resolve(__dirname),
          verbose: true,
        },
      ),
      new webpack.DefinePlugin({
        'GOOGLE_MAPS_API_KEY': JSON.stringify(googleMapsApiKey),
      }),
      //new webpack.optimize.UglifyJsPlugin(),
      new webpack.IgnorePlugin(/^\.\/locale$/, [ /moment$/, /react-intl$/, /zxcvbn$/ ]),
      new CopyWebpackPlugin(
        [
          {
            from: path.resolve(__dirname, 'public'),
            to: path.resolve(__dirname, 'dist'),
          }
        ],
        {
          ignore: [
            '*.html',
            '.*',
          ],
        },
      ),
      new FaviconsWebpackPlugin({
        logo: path.resolve(__dirname, 'public/favicon.jpg'),
        inject: true,
        title: 'Wonderful App',
      }),
      new webpack.optimize.ModuleConcatenationPlugin(),
      new webpack.optimize.SplitChunksPlugin({
        chunks: 'async',
        cacheGroups: {
          vendors: {
            test: /[\\/]node_modules[\\/]/,
            priority: -10,
          },
        },
      }),
      new webpack.HashedModuleIdsPlugin({
        hashFunction: 'sha256',
        hashDigest: 'hex',
        hashDigestLength: 20,
      }),
      new UglifyJsPlugin({
        uglifyOptions: {
          beautify: false,
          comments: false,
          comparisons: true,
          compress: true,
          conditionals: true,
          dead_code: true,
          evaluate: true,
          ie8: false,
          if_return: true,
          join_vars: true,
          keep_classnames: false,
          mangle: true,
          sequences: true,
          unused: true,
          warnings: false,
          output: {
            comments: false,
            beautify: false,
          },
        },
        exclude: [ /\.min\.js$/gi ], // skip pre-minified libs
      }),
      new CompressionPlugin({
        asset: '[path].gz[query]',
        algorithm: 'gzip',
        test: /\.js$|\.css$|\.html$/,
        threshold: 10240,
        minRatio: 0
      }),
    ],
  }
);

```

**Comment by ovaldi** [comment]
+1

**Comment by qasimalyas** [comment]
I'm also having this issue too

**Comment by wjurmy** [comment]
Same issue.

**Comment by ilinyhsergey** [comment]
The same issue after global ng-cli update

**Comment by evilebottnawi** [comment]
Please add screenshot devtool, other it is network problem

**Comment by ilinyhsergey** [comment]
![image](https://user-images.githubusercontent.com/4150029/43958878-99d32d88-9cb5-11e8-8e06-49ba2f353704.png)

![image](https://user-images.githubusercontent.com/4150029/43958979-d7fda69c-9cb5-11e8-8cad-158cec270b39.png)

Other lazy loaded modultes work fine

**Comment by ilinyhsergey** [comment]
In my case it was browser adblock plugin. The required module have 'advertising' in it's name.

**Comment by JZLeung** [comment]
Same issue.
And I found that it often appears on several models, like OPPO R9s, VIVO x7 and so on.

**Comment by RichPC** [comment]
I was seeing the same error with Laravel Mix (Webpack 3)
in my config I changed

`chunkFilename: '[name].[chunkhash].js',`

to

`chunkFilename: '[name][chunkhash].js',`

That fixed my broken site, but I've yet to see if the next deploy will reintroduce the cache/chunk loading problem.

**Comment by Serginho** [comment]
Same issue.

**Comment by realalexhomer** [comment]
We are seeing this issue in our sentry from iOs mobile safari users. It doesn't seem related to the version of iOs or safari. It certainly seems like an issue with webpack itself.

Its very hard to debug as its not happening consistently. Here is our webpack config / dependencies in package.json, hope it helps:

```
const path = require('path');
const webpack = require('webpack');
const CompressionPlugin = require('compression-webpack-plugin');
const SentryCliPlugin = require('@sentry/webpack-plugin');
const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
var StatsPlugin = require('stats-webpack-plugin');

module.exports = {
  name: 'client',
  target: 'web',
  entry: ['regenerator-runtime/runtime.js', path.resolve(__dirname, '../src/index.tsx')],
  output: {
    filename: '[name].[chunkhash].js',
    chunkFilename: '[name].[chunkhash].js',
    path: path.resolve(__dirname, '../dist/client'),
    publicPath: '/static/'
  },
  devtool: false,
  stats: 'normal',
  module: {
    rules: [
      {
        test: [ /\.ts$/, /\.tsx$/, /\.js$/ ],
        exclude: /node_modules/,
        loader: 'babel-loader',
        options: {
          presets: [
            [
              '@babel/env'
            ]
          ]
        }
      },
      {
        test: /\.(graphql|gql)$/,
        exclude: /node_modules/,
        use: [
          {
            loader: 'graphql-tag/loader'
          }
        ]
      }
    ]
  },
  mode: 'production',
  resolve: {
    modules: [ path.resolve(__dirname, '../src'), 'node_modules' ],
    extensions: ['.mjs', '.ts', '.tsx', '.js', '.graphql', '.gql']
  },
  plugins: [
    new ForkTsCheckerWebpackPlugin({
      async: false
    }),
    new StatsPlugin('stats.json', {
      chunkModules: true
    }),
    new webpack.optimize.ModuleConcatenationPlugin(),
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HashedModuleIdsPlugin(),
    new CompressionPlugin({
      filename: '[path].gz[query]',
      algorithm: 'gzip',
      test: /\.js$/,
      minRatio: 0.8
    }),
    new BundleAnalyzerPlugin({
      analyzerMode: 'static'
    }),
    new webpack.SourceMapDevToolPlugin({
      append: false,
      filename: '../maps/[file].map'
    }),
    new SentryCliPlugin({
      release: process.env.HEROKU_RELEASE_VERSION,
      include: path.resolve(__dirname, '../dist/client'),
      validate: true,
      dryRun: !process.env.SENTRY_AUTH_TOKEN
    }),
  ]
};
```

```
  "dependencies": {
    "@babel/polyfill": "^7.0.0",
    "@dubsmash/dubsmash-tracking": "1.1.9",
    "@sentry/browser": "^4.0.6",
    "@types/classnames": "^2.2.6",
    "@types/react-swipeable": "^4.2.2",
    "abort-controller": "^1.0.2",
    "apollo": "^1.6.0",
    "apollo-cache-inmemory": "^1.3.6",
    "apollo-client": "^2.4.3",
    "apollo-link": "^1.2.3",
    "apollo-link-error": "^1.1.1",
    "apollo-link-http": "^1.5.5",
    "apollo-link-state": "^0.4.2",
    "btoa": "^1.2.1",
    "classnames": "^2.2.6",
    "clipboard-polyfill": "^2.5.4",
    "compression": "^1.7.3",
    "express-static-gzip": "^1.0.0",
    "graphql": "^14.0.2",
    "history": "^4.7.2",
    "isomorphic-fetch": "^2.2.1",
    "js-cookie": "^2.2.0",
    "lodash.debounce": "^4.0.8",
    "lodash.merge": "^4.6.1",
    "querystring": "^0.2.0",
    "raven": "^2.6.3",
    "react": "^16.5.2",
    "react-adopt": "^0.6.0",
    "react-apollo": "^2.1.9",
    "react-dom": "^16.5.2",
    "react-helmet": "^5.2.0",
    "react-hot-loader": "^4.3.4",
    "react-jss": "^8.6.1",
    "react-router": "^4.3.1",
    "react-router-dom": "^4.3.1",
    "react-swipeable": "^4.3.0",
    "react-tree-walker": "^4.2.0",
    "reset-jss": "^1.0.0",
    "srt-to-vtt": "^1.1.3",
    "uuid": "^3.3.2",
    "webpack-flush-chunks": "^2.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.1.2",
    "@babel/plugin-proposal-class-properties": "^7.1.0",
    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
    "@babel/preset-env": "^7.1.0",
    "@babel/preset-react": "^7.0.0",
    "@babel/preset-typescript": "^7.1.0",
    "@sentry/webpack-plugin": "^1.6.1",
    "@types/btoa": "^1.2.0",
    "@types/enzyme": "^3.1.13",
    "@types/enzyme-adapter-react-16": "^1.0.3",
    "@types/enzyme-to-json": "^1.5.2",
    "@types/express": "^4.16.0",
    "@types/google.analytics": "^0.0.39",
    "@types/history": "^4.6.2",
    "@types/isomorphic-fetch": "^0.0.34",
    "@types/jest": "^23.3.1",
    "@types/js-cookie": "^2.2.0",
    "@types/jss": "^9.5.6",
    "@types/lodash.debounce": "^4.0.4",
    "@types/lodash.merge": "^4.6.4",
    "@types/query-string": "^6.1.0",
    "@types/raven": "^2.5.1",
    "@types/react": "^16.4.7",
    "@types/react-dom": "^16.0.6",
    "@types/react-helmet": "^5.0.6",
    "@types/react-router": "^4.0.30",
    "@types/react-router-dom": "^4.3.0",
    "@types/uuid": "^3.4.3",
    "@types/webpack-env": "^1.13.6",
    "apollo-codegen": "^0.20.2",
    "apollo-link-schema": "^1.1.0",
    "array.prototype.flatmap": "^1.2.1",
    "babel-loader": "^8.0.4",
    "commander": "^2.17.1",
    "compression-webpack-plugin": "^2.0.0",
    "cookie-parser": "^1.4.3",
    "cross-env": "^5.2.0",
    "enzyme": "^3.4.4",
    "enzyme-adapter-react-16": "^1.2.0",
    "enzyme-to-json": "^3.3.4",
    "express": "^4.16.3",
    "fork-ts-checker-webpack-plugin": "^0.4.9",
    "gaze": "^1.1.3",
    "get-graphql-schema": "^2.1.1",
    "glob": "^7.1.2",
    "gql2ts": "^1.9.0",
    "graphql-tools": "^3.1.1",
    "helmet": "^3.13.0",
    "html-webpack-plugin": "^3.2.0",
    "html-webpack-template": "^6.2.0",
    "husky": "^0.14.3",
    "jest": "^23.5.0",
    "jest-transform-graphql": "^2.1.0",
    "jss-snapshot-serializer": "^1.0.0",
    "lint-staged": "7.0.5",
    "merge-graphql-schemas": "^1.5.3",
    "morgan": "^1.9.0",
    "nightwatch": "^0.9.21",
    "pm2": "^3.2.2",
    "regenerator-runtime": "^0.12.0",
    "source-map-loader": "^0.2.3",
    "stats-webpack-plugin": "^0.7.0",
    "ts-jest": "^23.1.4",
    "ts-loader": "^5.1.0",
    "tslint": "^5.11.0",
    "tslint-eslint-rules": "^5.3.1",
    "tslint-react": "^3.6.0",
    "typescript": "^3.0.1",
    "uglifyjs-webpack-plugin": "^1.2.7",
    "webpack": "^4.16.2",
    "webpack-bundle-analyzer": "^3.0.2",
    "webpack-cli": "^3.1.0",
    "webpack-dev-middleware": "^3.1.3",
    "webpack-dev-server": "^3.1.5",
    "webpack-hot-middleware": "^2.22.3",
    "webpack-hot-server-middleware": "^0.5.0",
    "write-file-webpack-plugin": "^4.3.2"
  },
```

**Comment by camolin3** [comment]
This happens to my when running my server locally... when interacting with the app after killing my local server.

In DevTools, notice which is the file could not be loaded, and its error. You might get a better idea of what the problem is: file is not found, you do not have access, etc.

**Comment by piehei** [comment]
I'm getting this error from both iOS Safari and Chrome for Android users.
Is there some info I could provide to help diagnose this?

One thing to note: I'm getting this both for js and css files.

**Comment by tzimmermann** [comment]
We finally figured out that in our case, webpack was not the problem at all, but merely a symptom:

All our resources, including the webpack bundle files, need to pass through an intermediate proxy servlet in our Java backend (don't ask why).
This servlet had a thread-safety issue that only surfaced for parallel requests: Sometimes, one request for, say, a `foo.css` file would get a `bar.bundle.js` file as a response (and vice-versa).
This would then produce the `Loading chunk # failed` error message, as the webpack bundle file could not be parsed (because it actually contained a .css file).

As we used webpack prefetching more extensively, these errors became more frequent, since the browser tends to prefetch a lot of resources in parallel, apparently.

**Comment by js2me** [comment]
> I was seeing the same error with Laravel Mix (Webpack 3)
> in my config I changed
> 
> `chunkFilename: '[name].[chunkhash].js',`
> 
> to
> 
> `chunkFilename: '[name][chunkhash].js',`
> 
> That fixed my broken site, but I've yet to see if the next deploy will reintroduce the cache/chunk loading problem.

@RichPC Is that fixed this problem? Or you still have sometimes error with loading chunks?

**Comment by jacobrask** [comment]
This happens if a chunk could not be loaded, which will happen more often on mobile due to poor connections. You need to have some error handling in case a chunk failed to load, such as asking the user to reload the page.

**Comment by RichPC** [comment]
@js2me my production builds have remained free of that bug since I made that change.

@jacobrask Thanks for the tip, I'll look into that, but I was seeing the error on a wifi and fibre connection. The chunk filename format change fixed it for me.

**Comment by anton164** [comment]
@jacobrask, what's a good way of catching a "failed loading chunk" error?

From what I can tell, this is internal to webpack. I tried adding an `onerror` to the script tag that loads my JS bundle, but that isn't triggered

**Comment by jacobrask** [comment]
There might be different errors here for different people, but if you have a dynamic `import()` you should make sure to have an error handler, like `import('./chunk').then(renderSomething, showErrorMessage)`

**Comment by anton164** [comment]
Thanks @jacobrask I seem to have tracked down the error to these lines:
```
require(['license'], () => {
  require([
    'jqueryui',
    'jquery-browser-plugin',
    'lib-adapters/bootstrap-adapter',
    'lib-adapters/tablesorter-adapter',
    'lib-adapters/backbone-adapter'
  ], () => {
    require([
      'ardoq.app'
    ], (AQ) => {
      AQ.start();
    });
  });
});
```

^ We have an old project using AMD syntax, so there's a bunch of requires. After reading last night, it seems that this might cause webpack to code split?

**Comment by royipressburger** [comment]
Im having the exact issue with my react application that sits on AWS S3 as a static web host,
If the user is on some page in the application, lets say chunk number 1, and then We deploy a new version. when the user navigate a different page (different chunk) he will get this error.

Did anyone mange to solve this and still use chunks?

**Comment by realalexhomer** [comment]
@royipressburger this sounds like a different issue as its related to deployments for you. I have run into problems in the past where clients will still know about an older bundle, but we removed the old bundle on deployment. The client will try to access the old bundle and you will see this issue.

I would recommend seeing if you can increase the life of the old bundles for a little bit longer to alleviate this.

**Comment by jacobrask** [comment]
@royipressburger you can either

A. Keep old deploys around the server
B. Download and cache all chunks in the background on app launch either with `<link rel=prefetch/preload>` or service workers
C. Show an error message if a chunk load failed, asking the user to reload the app

Also make sure your webpack config is correct to make sure chunk hashes don't update if the content didn't change, to minimize the problem.

**Comment by royipressburger** [comment]
@realalexhomer @jacobrask Thanks guys, I think I will keep in the server always one version back. that way I reduce the chances of it to happen (still can happen if we push 2 new versions in small window of time but the chances are like 0.1%).

by the way @jacobrask how can catch that kind of error so ill be able to pop a message?

**Comment by nthompson777** [comment]
Just experienced the same problem with the 'Missing Chunk' Console Error in Chrome. This was a lazy loaded component that I recently made changes to and was unable to access it in my browser to verify the view. Was going a bit nuts trying to track down the problem when all I had to do was do a 'Empty Cache and Hard Reload' in Chrome (even though I already had my Dev Tools open with Cache disabled) which solved my 'Missing Chunk' issue.

**Comment by MZanggl** [comment]
(using Laravel-Mix)

Having the same issue here with a SPA. But isn't it natural for a SPA? The `manifest.js` which holds the files and hashes doesn't update without refreshing and when it loads a chunk it obviously uses the old hash from manifest.js.

Why are we doing
`chunkFilename: '[name].[chunkhash].js',`
in the first place, instead of just
`chunkFilename: '[name].js?id=[chunkhash]',`

This way it would at the very least get the cached version. Or am I missing something here? This is how I load all my non-codesplitted files and it's working like a charm.

The only problem I still see is when there was a new chunk added that doesn't exist in `manifest.js` yet.

**Comment by js2me** [comment]
It is network problem. Better solution is not use lazy load modules 😊


Sincerely, Volkov Sergey

Software Designer

Nizhny Novgorod, Russia

site: http://svolkov.pw/

github: js2me<https://github.com/js2me>

twitter: @volkovsergei96<https://twitter.com/volkovsergei96>





________________________________
От: 离上的风 <notifications@github.com>
Отправлено: Monday, December 3, 2018 7:28:45 AM
Кому: webpack/webpack
Копия: Sergey S. Volkov; Mention
Тема: Re: [webpack/webpack] Uncaught (in promise) Error: Loading chunk # failed. (#7502)


Have any methods for it?I have same issue

**Comment by evilebottnawi** [comment]
Closed. Lock conversation to avoid spamming (please use `stackoverflow` or `gitter` for questions and discussions). If somebody can reproduce problem and create minimum reproducible test repo feel free to open new issue with link on this repo. 

Also it can be happens with Service Worker and invalid setup.

Thanks for using `webpack`!
